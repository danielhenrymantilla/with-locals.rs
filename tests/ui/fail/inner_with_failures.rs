include!("../prelude.rs");

#[with]
fn stmt ()
{
    #[with]
    print!();
}

#[with]
fn expr ()
{
    let _ = #[with] expr();
}

#[with]
fn uncomplete_let ()
{
    #[with]
    let _incomplete;
}

#[with]
fn bad_rhs_of_let_not_a_function_name ()
{
    #[with]
    let _ = {foo}();
}

#[with]
fn bad_rhs_of_let_not_even_a_function_call ()
{
    #[with]
    let _ = 42;
}

#[with]
fn bad_rhs_has_attr ()
{
    #[with]
    let _ = #[extraneous] foo();
}
