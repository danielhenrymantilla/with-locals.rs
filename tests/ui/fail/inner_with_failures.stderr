error: `#[with]` does not support attributes
  --> $DIR/inner_with_failures.rs:41:13
   |
41 |     let _ = #[extraneous] foo();
   |             ^

error: `function(...)` or `<expr>.method(...)` expected
  --> $DIR/inner_with_failures.rs:34:13
   |
34 |     let _ = 42;
   |             ^^

error: Expected a function name
  --> $DIR/inner_with_failures.rs:27:13
   |
27 |     let _ = {foo}();
   |             ^^^^^

error: Missing expression
  --> $DIR/inner_with_failures.rs:20:20
   |
20 |     let _incomplete;
   |                    ^

error: `#[with]` must be applied to a `let` binding.
  --> $DIR/inner_with_failures.rs:13:13
   |
13 |     let _ = #[with] expr();
   |             ^

error: `#[with]` must be applied to a `let` binding.
 --> $DIR/inner_with_failures.rs:6:5
  |
6 |     #[with]
  |     ^
